src/graphql/resolvers.js

```diff
@@ -58,6 +99,78 @@ export default {
         discountCode: errors.length === 0 ? code : null,
         errors
       };
+    },
+
+    createContributor: async (_, { input }) => {
+      const shopifyCustomerID = await createShopifyCustomer(input);
+
+      const [
+        { totalContributions: contributionCount, pullRequests }
+      ] = await Promise.all([
+        getContributorInfo(input.githubUsername),
+        prisma.createContributor({
+          githubUsername: input.githubUsername,
+          email: input.email,
+          shopifyCustomerID
+        })
+      ]).catch(error => {
+        // @todo: more useful error logging
+        throw new ApolloError(error.message);
+      });
+
+      return {
+        githubUsername: input.githubUsername,
+        email: input.email,
+        shopifyCustomerID,
+        github: {
+          username: input.githubUsername,
+          contributionCount,
+          pullRequests
+        }
+      };
+    },
+    addTagsToShopifyCustomer: async (_, { id, tags }) => {
+      return await addTagsToCustomer(id, tags);
+    }
+  },
+  Contributor: {
+    shopify: async data => {
+      if (!data || !data.shopifyCustomerID) {
+        return null;
+      }
+
+      return {
+        id: data.shopifyCustomerID,
+        // Passed through to child resolver at `ShopifyInfo.codes`
+        count: data.github.contributionCount
+      };
+    }
+  },
+  ShopifyInfo: {
+    codes: async data => {
+      try {
+        return await getShopifyDiscountCodes(data.id, data.count);
+      } catch (error) {
+        // @todo: more useful error logging
+        throw new ApolloError(error.message);
+      }
+    },
+    tags: async data => {
+      try {
+        const customer = await getShopifyCustomer(data.id);
```

Comment:
I'm thinking we should memoize the `getShopifyCustomer` calls. I _think_, since this runs on Lambda, that the memoization would be short-lived, so it would pretty much just prevent us from hitting issues with making this call to Shopify multiple times during the operations that return `ShopifyInfo`.

TODO: research this.